src/hooks/useNfc.ts

```diff
@@ -17,6 +17,29 @@ export const useNfc = () => {
   const [isNfcNeedsTap, setIsNfcNeedsTap] = useState(false);
   const [isNfcNeedsPermission, setIsNfcNeedsPermission] = useState(false);
   const [isNfcActionSuccess, setIsNfcActionSuccess] = useState(false);
+  const [isPinRequired, setIsPinRequired] = useState(false);
+  const [isPinConfirmed, setIsPinConfirmed] = useState(false);
+
+  const [ pinResolver, setPinResolver ] = useState({ resolve: undefined })
+
+  const setPin = (pin: string) => {
+    if(pinResolver.resolve) {
+      pinResolver.resolve(pin);
+    }
+  }
+
+  const getPin = useCallback(async () => {
+    const pinInputPromise = () => {
+      let _pinResolver;
+      return [ new Promise(( resolve ) => {
+          _pinResolver = resolve
+      }), _pinResolver]
+    }
+
+    const [ promise, resolve ] = pinInputPromise()
+    setPinResolver({ resolve })
```

Comment:
You can fix TS issues here by committing this

```suggestion
  const [pinResolver, setPinResolver] = useState<{
    resolve: (v: string) => void;
  }>();

  const setPin = (pin: string) => {
    if (pinResolver?.resolve) {
      pinResolver.resolve(pin);
    }
  };

  const getPin = useCallback(async () => {
    const pinInputPromise = () => {
      let _pinResolver;
      return [
        new Promise<string>((resolve) => {
          _pinResolver = resolve;
        }),
        _pinResolver
      ];
    };

    const [promise, resolve] = pinInputPromise();
    setPinResolver({ resolve } as unknown as { resolve: (v: string) => void });
```

Reply:
I had something like that but wasn't thinkin about casting first to unknown. Thanks 👍